robotoff/workers/tasks/update_recycle.py

```
@@ -0,0 +1,206 @@
+                    logger.info("Recycle image(selected): ", ean, field, "->", image)
+
+
+def check_image_in_pack(image, lang, pack_images):
+    """
+    Check if image has been already selected
+
+    :param image: imgid
+    :param lang: en/fr/...
+    :param pack_images: image ids [packaging_en, front_en]
+    :return:
+    """
+    if "packaging_{}".format(lang) in pack_images:
+        current_id = pack_images.get("packaging_{}".format(lang), {}).get("imgid")
+        if current_id == image:
+            return True
+
+    return False
+
+
+def unselect_image(barcode, field_name, username, password):
+    """
+    Unselect image for product
+
+    :param barcode: ean
+    :param field_name: packaging_en
+    :param username:
+    :param password:
+    :return:
+    """
+
+    url = "https://world.openfoodfacts.org/cgi/product_image_unselect.pl"
+    data = {
+        "code": barcode,
+        "id": field_name,
+        "user_id": username,
+        "password": password,
+    }
+    try:
+        r = requests.post(url, data=data)
+        if r.ok:
+            return True
+    except Exception:
+        logger.warn(
+            "Exception in unselect_image: barcode - {}, id - {}".format(
+                barcode, field_name
+            )
+        )
+
+    return False
+
+
+def reselect_image(barcode, field_name, img_id, username, password):
+    """
+    Select image for product
+
+    :param barcode: ean
+    :param field_name: packaging_en
+    :param img_id: imgid=[0, 1, 2]
+    :param username:
+    :param password:
+    :return:
+    """
+
+    if img_id:
+        url = "https://world.openfoodfacts.org/cgi/product_image_crop.pl"
+        data = {
+            "code": barcode,
+            "imgid": img_id,
+            "id": field_name,
+            "user_id": username,
+            "password": password,
+        }
+        try:
+            r = requests.post(url, data=data)
+            if r.ok:
+                return True
+        except Exception:
```

Same here about the exception handling. Is it a good thing to ignore errors?

---

Same as above